python/pyiceberg/schema.py

```diff
@@ -273,6 +277,49 @@ def field_ids(self) -> Set[int]:
         """Returns the IDs of the current schema."""
         return set(self._name_to_id.values())
 
+    @staticmethod
+    def validate_identifier_field(field_id: int, id_to_field: Dict[int, NestedField], id_to_parent: Dict[int, int]) -> None:
+        """Validates that the field with the given ID is a valid identifier field.
+
+        Args:
+          field_id: The ID of the field to validate.
+          id_to_field: A map from field IDs to field objects.
+          id_to_parent: A map from field IDs to their parent field IDs.
+
+        Raises:
+          ValueError: If the field is not valid.
+        """
+        field = id_to_field.get(field_id)
+        if field is None:
+            raise ValueError(f"Cannot add fieldId {field_id} as an identifier field: field does not exist")
+
+        if not field.field_type.is_primitive:
+            raise ValueError(f"Cannot add field {field.name} as an identifier field: not a primitive type field")
+
+        if not field.required:
+            raise ValueError(f"Cannot add field {field.name} as an identifier field: not a required field")
+
+        if isinstance(field.field_type, DoubleType) or isinstance(field.field_type, FloatType):
```

Comment:
These can be combined:
```suggestion
        if isinstance(field.field_type, (DoubleType, FloatType)):
```